be/src/runtime/load_channel_mgr.cpp

```diff
@@ -85,23 +85,60 @@ LoadChannelMgr::~LoadChannelMgr() {
 }
 
 void LoadChannelMgr::close() {
-    std::lock_guard l(_lock);
-    for (auto iter = _load_channels.begin(); iter != _load_channels.end();) {
-        iter->second->cancel();
-        iter->second->abort();
-        iter = _load_channels.erase(iter);
+    {
+        std::lock_guard l(_lock);
+        for (auto iter = _load_channels.begin(); iter != _load_channels.end();) {
+            iter->second->cancel();
+            iter->second->abort();
+            iter = _load_channels.erase(iter);
+        }
+    }
+    if (_async_rpc_pool) {
+        _async_rpc_pool->shutdown();
     }
 }
 
 Status LoadChannelMgr::init(MemTracker* mem_tracker) {
     _mem_tracker = mem_tracker;
     RETURN_IF_ERROR(_start_bg_worker());
+    int num_threads = config::load_channel_rpc_thread_pool_num;
+    if (num_threads <= 0) {
+        num_threads = CpuInfo::num_cores();
+    }
+    RETURN_IF_ERROR(ThreadPoolBuilder("load_channel_mgr")
+                            .set_min_threads(5)
+                            .set_max_threads(num_threads)
+                            .set_max_queue_size(config::load_channel_rpc_thread_pool_queue_size)
+                            .set_idle_timeout(MonoDelta::FromMilliseconds(10000))
+                            .build(&_async_rpc_pool));
+    REGISTER_THREAD_POOL_METRICS(load_channel_mgr, _async_rpc_pool.get());
     return Status::OK();
 }
 
 void LoadChannelMgr::open(brpc::Controller* cntl, const PTabletWriterOpenRequest& request,
                           PTabletWriterOpenResult* response, google::protobuf::Closure* done) {
-    ClosureGuard done_guard(done);
+    LoadChannelOpenRequest open_request;
+    open_request.cntl = cntl;
+    open_request.request = &request;
+    open_request.response = response;
+    open_request.done = done;
+    open_request.receive_rpc_time_ns = MonotonicNanos();
+    if (!config::enable_load_channel_rpc_async) {
+        _open(open_request);
+        return;
+    }
+    auto task = [=]() { this->_open(open_request); };
+    Status status = _async_rpc_pool->submit_func(std::move(task));
```

Comment:
the task can be submit successful but not executed at all when doing pool shutdown, hence lead the `done` never get called and resulting the brpc server exit hanging.

Reply:
+1

Reply:
fixed. not shutdown the thread pool to ensure all tasks can be executed

Reply:
Worried about the thread pool execution may cause invalid access to global variables if the tasks are stilling running but the global env get destroyed during the exit. Although right now it is almost sure that the global exit process will be blocked there in brpc_server->join() and no further action can be taken if there is any rpc done references held here.

Reply:
how about to use destructor to guarantee the response, see https://github.com/StarRocks/starrocks/pull/56517/commits/d4b3c1728ed1e84cbc2696d4be900f09a4b46dee